src/sentry/monitors/clock_dispatch.py

```diff
@@ -70,6 +76,33 @@ def _dispatch_tick(ts: datetime):
     _clock_tick_producer.produce(ArroyoTopic(topic), payload)
 
 
+def _make_reference_ts(ts: datetime):
+    """
+    Produce a timestamp number with the seconds and microsecond removed
+    """
+    return int(ts.replace(second=0, microsecond=0).timestamp())
+
+
+def update_check_in_volume(ts: datetime):
+    """
+    Increment a counter for this particular timestamp trimmed down to the
+    minute.
+
+    This counter will be used as historical data to help incidate if we may
+    have had some data-loss (due to an incident) and would want to tick our
+    clock in a mode where misses and time-outs are created as "unknown".
+    """
+    redis_client = redis.redis_clusters.get(settings.SENTRY_MONITORS_REDIS_CLUSTER)
+
+    reference_ts = _make_reference_ts(ts)
+    key = MONITOR_VOLUME_HISTORY.format(reference_ts)
+
+    pipeline = redis_client.pipeline()
+    pipeline.incr(key, amount=1)
+    pipeline.expire(key, MONITOR_VOLUME_RETENTION)
```

Comment:
We could just probabilistically set the `expire` 10% of the time to nearly halve the number of writes. Probably no need if we do the batching option though

Reply:
Going to leave this as is since we have the bulk version now.